### ISSUE TITLE
HMR not working for Vue.js

### ISSUE THREAD

**Issue by lwakefield** [thereactivestack/meteor-webpack#99]
In adding Vue.js, it seems it was made so that HMR is only available for React.
Is there any particular reason for this?

Specifically this change: https://github.com/thereactivestack/meteor-webpack/commit/e4e8426415d0813304983e8d06f0b0cd80a3d5b0#diff-2ea8dd33416614d8f7039980b6915540R410

Also :+1: for adding Vue.js

**Comment by eXon** [comment]
I though at first vue-loader was taking care of this but it seems that I'm wrong. I'll remove that if.

**Comment by lwakefield** [comment]
Thank you very much!
I would have made a pull-request, but I wasn't sure if there was more to that if statement.

**Comment by eXon** [comment]
@lwakefield You can update to 1.0.11 and let me know if it works!

**Comment by lwakefield** [comment]
I seem to still have an issue, though it may or may not be related to this issue.

Just for a simple app with an empty `client.js` and `server.js` (so this is probably more related to the babel config), I get the following error:

    Module not found: Error: Cannot resolve module 'webpack-hot-middleware/client' in /Users/lawrence/Desktop/meteor-vue-test

I am still looking into this, but prior to 1.0.6 (addition of vue) I was using the following `webpack.config.js` for vue:

```
var weight = 100;

function dependencies() {
    return {
        devDependencies: {
            'vue-loader': '^8.2.0',
            'babel-loader' : '^6.2.0',
            'babel-preset-es2015': '^6.3.13'
        }
    };
}

function config(settings) {
    var babelSettings = {
        presets: ['es2015']
    };

    return {
        loaders: [
        { test: /\.js$/, loader: 'babel', query: babelSettings, exclude: /node_modules/ },
        { test: /\.vue$/, loader: 'vue' }
        ],
        extensions: ['.js', '.vue'],
    };
}
```

This still works, but I am not sure what the difference is yet.
Will keep you updated.

**Comment by eXon** [comment]
Have you tried `npm install webpack-hot-middleware`?

**Comment by lwakefield** [comment]
I have been able to fix this by changing the extensions in `webpack.config.js`:

```js
extensions: ['.js', '.vue']
```

Also, can confirm `webpack-hot-middleware` is installed and sitting in `node_modules`.

**Comment by eXon** [comment]
It should have been in your devDependencies of your package.json. Webpack is adding it automatically, that's weird.

**Comment by lwakefield** [comment]
I can create a new issue for this if you like, but I am pretty sure adding '.js' to the extensions in `webpack.config.js` should solve this.

Unfortunately my knowledge of WebPack is not enough to know why it causes this particular error.

**Comment by eXon** [comment]
It's already fix if you update webpack:vue to 1.0.1 :)

**Comment by lwakefield** [comment]
Perfect, working well now! :+1:
